CMakeLists.txt

```diff
@@ -1,19 +1,57 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.8)
 
-project(dbg_macro)
+project(dbg_macro LANGUAGES CXX)
 
-set(CMAKE_CXX_STANDARD 11)
+###############################################################################
+# CMake Options
+###############################################################################
+
+option(DBG_MACRO_BUILD_EXAMPLES OFF)
+option(DBG_MACRO_BUILD_TESTS OFF)
+
+###############################################################################
+# Add dbg_macro Library
+###############################################################################
 
 add_library(${PROJECT_NAME} INTERFACE)
-target_include_directories(${PROJECT_NAME} INTERFACE ".")
+add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
+target_include_directories(${PROJECT_NAME} INTERFACE
```

Comment:
do we really have to move the `dbg.h` header to a separate `include` folder? I know it's a good practice, but this will always be a single-header library and I think it would be nice to have the main file in the top-level directory of the  repo.

Reply:
I know this will always be a single header-only library. But look at `catch2` library and others. Those libraries exist in package managers and are available for installing. From my experience it is better to keep the header inside a directory because it makes `cmake install` easier and the file is constraint into its folder and not in the wild.

Reply:
Okay :+1:.

In this case, please make sure that all references to `./dbg.h` are updated (e.g. in the README)

Reply:
@sharkdp Roger That Sir!